### DISCUSSION TITLE
how do change SidecarProxyImage url

### DISCUSSION THREAD

**Discussion by yihaung** [istio/istio#49457] [Q&A]
We are building an istio in a closed network.
I created an image where the external Internet became available, and I "docker save" the image again in the closed network.

At this time, the envoy of the sidecar method is referring to the external repository.
However, I modified the image repository internally and confirmed that the container is also being serviced normally.

https://istio.io/latest/docs/reference/config/annotations/#SidecarProxyImage

I modified SidecarProxyImage url in a sidecar manner as I was guided on this page.
Modified and restarted in k8s deployment.

When you label and restart another service, the service still calls the external address "docker.io/istio/proxyv2.1.20.2 ".

In the restarted pod, announcements are well reflected in the metadata field.

Do I have to look at other parts?

**Comment by hanxiaop** [comment]
`sidecar.istio.io/proxyImage` is an annotation that overrides the proxy image when specified in the deployment template's annotations. If you also want global changes to the proxy image, you can try `values.global.proxy.image`.

**Reply by yihaung** [reply]
Thank you for your answer, it is generally not recommended to modify that option, so we solved it by adding internal repository to "--hub" in the installation option. Thank you.
